src/replication/replication_slave_node.cpp

```diff
@@ -140,4 +134,21 @@ namespace cubreplication
 
     return NO_ERROR;
   }
+
+  void slave_node::disconnect_from_master ()
+  {
+    delete m_transfer_receiver;
+    m_transfer_receiver = NULL;
+
+    m_stream_file->remove_sync_notifier ();
+
+    if (m_ctrl_sender != NULL)
+      {
+	m_ctrl_sender->stop ();
+	cubthread::get_manager ()->destroy_daemon_without_entry (m_ctrl_sender_daemon);
+	delete m_ctrl_sender;
```

Comment:
I don't think this is safe with m_lc not stopped (there might be a reference captured with log_consumer::set_ack_producer call).
Probably you need to call log_consumer::stop